libs/utils/env.py

```diff
@@ -439,6 +439,51 @@ def _init_target(self, force = False):
             self.__modules = ['bl', 'cpufreq']
             platform = Platform(model='pixel')
 
+        # Initialize gem5 platform
+        elif self.conf['board'].upper() == 'GEM5':
+            self.nrg_model = None
+            self.__modules=['cpufreq']
+
+            system_info = self.conf['gem5_settings']['system']
+            simulator_info = self.conf['gem5_settings']['simulator']
+            simulator_args = ''
+            platform_args = ''
+
+            # Get gem5 binary arguments
+            simulator_args = ' '.join(simulator_info.get('args', []) +
+                                      ['--listener-mode=on'])
+
+            # Get platform arguments
+            platform_args = ' '.join(system_info['platform'].get('args', []) +
+                                     ['--kernel {}'.format(system_info['kernel']),
+                                      '--dtb {}'.format(system_info['dtb']),
+                                      '--disk-image {}'.format(system_info['disk'])])
+
+            # Gather all arguments
+            args = ' '.join([simulator_args,
+                             system_info['platform']['description'],
+                             platform_args])
+
+            diod_path = which('diod')
+            if diod_path is None:
+                raise RuntimeError('Failed to find diod on your host machine, '
```

Comment:
Maybe better (notice the quotes):
`Failed to find "diod" on your host machine`...